src/FSharpLint.Core/Rules/Conventions/SuggestUseAutoProperty.fs

```diff
@@ -0,0 +1,99 @@
+                            match headPat with
+                            | SynPat.Named (_, bindingIdent, _, _, _) when isMutable ->
+                                bindingIdent.idText = ident.idText
+                            | _ -> false)
+                    | _ -> false)
+            | _ -> false
+
+        not isMutableVariable
+    | SynExpr.ArrayOrList (_, elements, _) ->
+        elements
+        |> List.forall (isImmutableValueExpression args)
+    | SynExpr.ArrayOrListOfSeqExpr (_, SynExpr.CompExpr(true, _, innerExpr, _), _) ->
+        isImmutableValueExpression args innerExpr
+        || isImmutableSequentialExpression args innerExpr
+    | _ -> false
+
+and isImmutableSequentialExpression args expression =
+    match expression with
+    | SynExpr.Sequential (_, _, expr1, expr2, _) ->
+        isImmutableValueExpression args expr1
+        && (isImmutableSequentialExpression args expr2
+            || isImmutableValueExpression args expr2)
+    | _ -> false
+
+let private runner (args: AstNodeRuleParams) =
+    match args.AstNode with
+    | MemberDefinition
+        (
+            SynMemberDefn.Member
+                (
+                    SynBinding
+                        (
+                            _accessibility,
+                            _kind,
+                            _,
+                            false,
+                            _attributes,
+                            _xmlDoc,
+                            SynValData (Some memberFlags, _, _),
+                            SynPat.LongIdent (memberIdentifier, _, _, argPats, _, _),
+                            _returnInfo,
+                            expr,
+                            _bindingRange,
+                            _
+                        ),
+                    memberRange
+                )
+        ) when memberFlags.IsInstance ->
+        match expr, argPats with
+        | _, SynArgPats.Pats pats when pats.Length > 0 -> // non-property member
+            Array.empty
+        | expression, _ when isImmutableValueExpression args expression ->
+            let suggestedFix =
+                lazy
+                    (match memberIdentifier.Lid with
+                     | [ _; memberName ] ->
+                         Some
+                             { FromText = args.FileContent
+                               FromRange = memberIdentifier.Range
+                               ToText = "val " + memberName.idText }
+                     | _ -> None)
+
+            { Range = memberRange
+              Message = Resources.GetString "RulesSuggestUseAutoProperty"
+              SuggestedFix = Some suggestedFix
+              TypeChecks = List.Empty }
+            |> Array.singleton
+        | _ -> Array.empty
+    | _ -> Array.empty
+
+let rule =
+    { Name = "SuggestUseAutoProperty"
+      Identifier = Identifiers.FavourConsistentThis
```

Comment:
@webwarrior-ws this is a typo here, please fix

Reply:
Fixed